src/app/search/page.tsx

```diff
@@ -0,0 +1,38 @@
+"use client";
+
+import { useProducts } from "@/app/hooks/useProducts";
+import SectionContainer from "../components/section-container";
+import ProductGrid from "../components/product-grid";
+
+export default function SearchPage({
+  searchParams,
+}: {
+  searchParams?: {
+    query?: string;
+    page?: string;
+  };
+}) {
+  const query = searchParams?.query || "";
+  const currentPage = Number(searchParams?.page) || 1;
+  const limit = 4;
```

Comment:
You can move this to global space to avoid the recreation of this variable on every render.